courseTasks/executor/executor.go

```diff
@@ -0,0 +1,103 @@
+package executor
+
+import (
+	"fmt"
+	"johnSamilin/golang-hangman/courseTasks/analyzer"
+	"johnSamilin/golang-hangman/courseTasks/executor/constants"
+	"johnSamilin/golang-hangman/courseTasks/utils"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+func startExecution(configId int, workers chan Worker, commands []string, exit chan bool, wg *sync.WaitGroup) analyzer.Stats {
+	var tasksPool = make(chan string, len(commands))
+	collector := analyzer.New()
+	for _, cmd := range commands {
+		tasksPool <- cmd
+		collector.AddLine(cmd)
+	}
+	for {
+		freeWorker := <-workers
+		if len(tasksPool) > 0 {
+			shouldStop := processCommand(freeWorker.Id, freeWorker.Lock, configId, <-tasksPool, exit)
+			if !shouldStop {
+				workers <- freeWorker
+			}
+			if shouldStop {
+				break
+			}
+		}
+		if len(tasksPool) == 0 {
+			close(tasksPool)
+			wg.Done()
+			break
+		}
+	}
+	return collector.GetStats()
+}
+
+func processCommand(idx int, lock *sync.Mutex, configId int, cmd string, exit chan bool) (shouldStop bool) {
+	shouldStop = false
```

Comment:
do not do it this way, it is very painful to extend.

Try this:

have a common interface for tasks (e.g. `Exec()` as the only method), then have a bunch of structs with different properties inside for different tasks. Then try to execute them here instead of writing the whole logic here.

Example:

```golang
type TimeTask struct {
  timeToSleep int
}

func (tt TimeTask) Exec() {
  time.Sleep(tt.timeToSleep)
}
```

this way the logic is abstracted away and here you have to worry about the execution order, concurrency and other important things (BUT NOT the business logic)